src/app/services/baw-api/audio-event-tags.service.ts

```diff
@@ -0,0 +1,34 @@
+import { HttpClient } from "@angular/common/http";
+import { Inject, Injectable } from "@angular/core";
+import { API_ROOT } from "@helpers/app-initializer/app-initializer";
+import { stringTemplate } from "@helpers/stringTemplate/stringTemplate";
+import { AudioEvent } from "@models/AudioEvent";
+import { AudioEventTag } from "@models/AudioEventTag";
+import { AudioRecording } from "@models/AudioRecording";
+import { Observable } from "rxjs";
+import { ApiFilter, Filter, id, IdOr, IdParam, option } from "./api-common";
+import { BawApiService, Filters } from "./baw-api.service";
+
+const audioRecordingId: IdParam<AudioRecording> = id;
+const audioEventId: IdParam<AudioEvent> = id;
```

Comment:
yes and no, i would prefer to extract the actual endpoint so that the following could happen:

```typescript
// Current
const audioRecordingId: IdParam<AudioRecording> = id;
const audioEventId: IdParam<AudioEvent> = id;
const endpoint_v1 = stringTemplate`/audio_recordings/${audioRecordingId}/audio_events/${audioEventId}/tags/${option}`;

// Future
import {parentPath} from './parent';
const audioEventId: IdParam<AudioEvent> = id;
const endpoint_v2 = stringTemplate`${parentPath}/audio_events/${audioEventId}/tags/${option}`;
```

Unfortunately, i can't think of a simple solution to this as i'm not familiar enough with template literals just yet. You're solution would be potentially more dangerous as there is a big difference between `IdParam` and `IdParamOptional` which would be hard to detect when the value is imported.

Reply:
That's something to look into, doesn't need to happen yet